src/mapex/write_parsed_mappings.py

```diff
@@ -204,69 +209,82 @@ def write_parsed_mappings_stix(parsed_mappings, filepath):
             stix_object["id"]
             for stix_object in stix_bundle["objects"]
             if stix_object.get("name") == mapping["capability_id"]
-        ][0]
+        ]
+        related_target_ref = technique_target_dict.get(mapping["attack_object_id"], "")
 
         # account for None mapping_type on not_mappable items
         mapping_type = (
             mapping["mapping_type"].replace("_", "-")
             if mapping["mapping_type"]
             else None
         )
-        stix_bundle["objects"].append(
-            {
-                "type": "relationship",
-                "id": f"relationship--{relationship_uuid}",
-                "spec_version": "2.1",
-                "created": datetime.now().isoformat(),
-                "modified": datetime.now().isoformat(),
-                "relationship_type": mapping_type,
-                "source_ref": related_source_ref,
-                "target_ref": technique_target_dict.get(
-                    mapping["attack_object_id"], ""
-                ),
-            },
+        # do not add a relationship node for a non-mappable technique
+        if (
+            related_source_ref
+            and related_target_ref
+            and not mapping_type == "non-mappable"
+        ):
+            stix_bundle["objects"].append(
+                {
+                    "type": "relationship",
+                    "id": f"relationship--{relationship_uuid}",
+                    "spec_version": "2.1",
+                    "created": created_date,
+                    "modified": created_date,
+                    "relationship_type": mapping_type,
+                    "source_ref": related_source_ref[0],
+                    "target_ref": related_target_ref,
+                },
+            )
+    # only write to file if stix is valid
+    validation_results = validate_instance(stix_bundle)
+    if validation_results.is_valid:
+        filepath = f"{filepath}_stix"
+        filepath_with_count = filepath
+        counter = 0
+        while os.path.exists(f"{filepath_with_count}.json"):
+            counter += 1
+            filepath_with_count = f"{filepath}_{counter}"
+        stix_file = open(
+            f"{filepath_with_count}.json",
+            "w",
+            encoding="UTF-8",
+        )
+        json.dump(stix_bundle, fp=stix_file)
+        logger.info(
+            "Successfully wrote mappings stix file to {filepath_with_count}.json",
+            filepath_with_count=filepath_with_count,
+        )
+    else:
+        logger.error(
```

Comment:
If we skip the file and log an error, it's unlikely to be noticed when the GitHub action runs. Then when the site is deployed, we'll just end up with a dead link, right?

If that's the case, it's better to throw an exception and fail the build. That way we will catch the mistake faster, i.e. probably in the PR.